# PERL and LUA setup for generating 64-bit binaries
# Not for upstream, Solaris-specific

--- rrdtool-1.6.0/configure.ac  2016-05-19 16:38:31.633135607 -0700
+++ rrdtool-1.6.0/configure.ac  2016-05-20 12:59:35.178966221 -0700
@@ -614,8 +614,6 @@
 AM_CONDITIONAL(ENABLE_RPATH,[test "x$enable_rpath" = "xyes"])
 
 dnl Check for Perl and friends
-PATH=$PATH:/usr/perl5/bin
-export PATH
 AC_PATH_PROG(PERL, perl, no)
 AC_PATH_PROG(POD2MAN, pod2man, no)
 AC_PATH_PROG(POD2HTML, pod2html, no)
@@ -666,7 +664,7 @@
 test "$langpref" = '$(DESTDIR)NONE' && langpref='$(DESTDIR)'$ac_default_prefix
 test "$langpref" = "NONE" && langpref=$ac_default_prefix
 
-PERL_MAKE_OPTIONS="PREFIX=$langpref INSTALL_BASE= LIB=$langpref/lib/perl/$PERL_VERSION"
+PERL_MAKE_OPTIONS="PREFIX=$langpref"
 
 dnl pass additional perl options when generating Makefile from Makefile.PL
 AC_ARG_ENABLE(perl-site-install,
@@ -688,6 +686,22 @@
    if test ! -z "$PERLLDFLAGS"; then
        PERL_MAKE_OPTIONS="$PERL_MAKE_OPTIONS LDFLAGS=$PERLLDFLAGS"
    fi
+
+   if test ! -z "$PERLLIB"; then
+       PERL_MAKE_OPTIONS="$PERL_MAKE_OPTIONS LIB=$PERLLIB"
+   fi
+
+   if test ! -z "$PERLINSTALLMAN3DIR"; then
+        PERL_MAKE_OPTIONS="$PERL_MAKE_OPTIONS INSTALLMAN3DIR=$PERLINSTALLMAN3DIR"
+   fi
+
+   if test ! -z "$PERLINSTALLSITEMAN3DIR"; then
+        PERL_MAKE_OPTIONS="$PERL_MAKE_OPTIONS INSTALLSITEMAN3DIR=$PERLINSTALLSITEMAN3DIR"
+   fi
+
+   if test ! -z "$PERLINSTALLVENDORMAN3DIR"; then
+        PERL_MAKE_OPTIONS="$PERL_MAKE_OPTIONS INSTALLVENDORMAN3DIR=$PERLINSTALLVENDORMAN3DIR"
+   fi
 fi
 
 AC_ARG_WITH(perl-options,
@@ -833,7 +847,12 @@
           done
         fi
 
-        LUA_RRD_LIBDIR="$libdir/lua/$lua_vdot"
+	 if test ! -z "$LUARRDLIBDIR"; then
+          LUA_RRD_LIBDIR="$langpref/lib/lua/$lua_vdot/$LUARRDLIBDIR"
+        else
+          LUA_RRD_LIBDIR="$langpref/lib/lua/$lua_vdot"
+        fi
+
         # if lua 5.0 can't find compat-5.1, force installation of
         # compat-5.1.lua together with RRDtool.
         if test "$lua_vdot" = "5.0" -a "$LUA_HAVE_COMPAT51" != "HAVE_COMPAT51"; then
